src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2300,15 +2186,13 @@ public bool DefaultReadOnly
 		public bool IsReadOnly(object entityOrProxy)
 		{
 			ErrorIfClosed();
-			// CheckTransactionSynchStatus();
 			return persistenceContext.IsReadOnly(entityOrProxy);
```

Comment:
I have hesitated to put this in a `BeginProcess` like the `BestGuessEntityName`. For scenario checking a list of entities, it could have a performance impact, so I have left it as is.